checks/src/main/kotlin/com/faithlife/lint/ForEachFunctionDetector.kt

```diff
@@ -10,23 +10,48 @@ import com.android.tools.lint.detector.api.Scope
 import com.android.tools.lint.detector.api.Severity
 import com.android.tools.lint.detector.api.SourceCodeScanner
 import com.intellij.psi.PsiMethod
+import org.jetbrains.kotlin.psi.KtLambdaExpression
 import org.jetbrains.uast.UCallExpression
 import org.jetbrains.uast.kotlin.KotlinUSafeQualifiedExpression
 
 class ForEachFunctionDetector : Detector(), SourceCodeScanner {
-    override fun getApplicableMethodNames(): List<String> = listOf("forEach", "forEachIndexed")
+    override fun getApplicableMethodNames(): List<String> =
+        listOf(FOR_EACH_NAME, FOR_EACH_INDEXED_NAME)
 
     override fun visitMethodCall(context: JavaContext, node: UCallExpression, method: PsiMethod) {
         if (node.uastParent is KotlinUSafeQualifiedExpression) return
 
+        val receiver = node.receiver
         val receiverTypeClass = context.evaluator.getTypeClass(node.receiverType)
 
-        if (receiverTypeClass != null && context.evaluator.inheritsFrom(receiverTypeClass, "java.lang.Iterable", false)) {
+        if (receiver != null && receiverTypeClass != null && context.evaluator.inheritsFrom(receiverTypeClass, "java.lang.Iterable", false)) {
```

Comment:
The `receiver != null` condition means that we won't detect any uses of the function with an implicit receiver. There should be a test for that, but alas.